File: src/nylon/parser/DefaultParser.java
package nylon.parser;

import nylon.nylonobjects.NylonFunction;

import java.util.ArrayList;

/**
 * Created by Aedan Smith.
 */

class DefaultParser {
    public static void addTo(ArrayList<Parser> parsers) {
        for (int i = 0; i < 256; i++) {
            parsers.add(i, DefaultParser::parse);
        }
    }

    public static NylonFunction parse(StringIterator in, NylonParser nylonParser) {
        in.skip();
        return null;
    }
}

File: src/nylon/parser/parsers/DefaultParser.kt
package nylon.parser.parsers

import nylon.NylonFunction
import nylon.parser.CharIterator
import nylon.parser.NylonParser
import nylon.parser.Parser
import nylon.parser.ParserBuilder
import java.util.*

/**
 * Created by Aedan Smith.
 */

class DefaultParserBuilder : ParserBuilder {
    override fun accept(parsers: HashMap<Char, Parser>) {
        for (i in 0..255) {
            parsers.put(i.toChar(), DefaultParser())
        }
    }
}

class DefaultParser : Parser {
    override fun apply(src: CharIterator, parser: NylonParser): NylonFunction? {
        src.next()
        return null
    }
}
